sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java

```diff
@@ -2780,31 +2820,54 @@ public void deleteCaseInExternalSurveillanceTool(Case caze) throws ExternalSurve
 	@Override
 	@RightsAllowed(UserRight._CASE_ARCHIVE)
 	public void archive(String entityUuid, Date endOfProcessingDate, boolean includeContacts) {
-		super.archive(entityUuid, endOfProcessingDate);
+		ProcessedEntity processedEntity = super.archive(entityUuid, endOfProcessingDate);
 		if (includeContacts) {
 			List<String> caseContacts = contactService.getAllUuidsByCaseUuids(Collections.singletonList(entityUuid));
 			contactService.archive(caseContacts);
 		}
+
+		if (processedEntity.getProcessedEntityStatus().equals(ProcessedEntityStatus.EXTERNAL_SURVEILLANCE_FAILURE)) {
+			throw new ExternalSurveillanceToolRuntimeException(
+				I18nProperties.getString(Strings.ExternalSurveillanceToolGateway_notificationErrorArchiving));
+		}
 	}
 
 	@Override
 	@RightsAllowed(UserRight._CASE_ARCHIVE)
-	public void archive(List<String> entityUuids, boolean includeContacts) {
-		super.archive(entityUuids);
+	public List<ProcessedEntity> archive(List<String> entityUuids, boolean includeContacts) {
+		List<ProcessedEntity> processedEntities = super.archive(entityUuids);
 		if (includeContacts) {
 			List<String> caseContacts = contactService.getAllUuidsByCaseUuids(entityUuids);
 			contactService.archive(caseContacts);
 		}
+
+		return processedEntities;
 	}
 
 	@Override
 	@RightsAllowed(UserRight._CASE_ARCHIVE)
-	public void dearchive(List<String> entityUuids, String dearchiveReason, boolean includeContacts) {
-		super.dearchive(entityUuids, dearchiveReason);
+	public ProcessedEntity dearchive(String entityUuid, String dearchiveReason, boolean includeContacts) {
+		ProcessedEntity processedEntity = dearchive(Collections.singletonList(entityUuid), dearchiveReason, includeContacts).get(0);
+
+		if (processedEntity.getProcessedEntityStatus().equals(ProcessedEntityStatus.EXTERNAL_SURVEILLANCE_FAILURE)) {
+			throw new ExternalSurveillanceToolRuntimeException(
```

Comment:
Yes, I threw an exception instead of handling the status because I've thought that it would be more consistent with the existent implementation for Infrastructure entities when an Access Denied will be thrown. I've changed now for Cases and Events. Should I refactor for infrastructure entities too?

![image](https://github.com/SORMAS-Foundation/SORMAS-Project/assets/47103965/4acffa29-65e8-49e2-8b2f-d57dbf9ad356)

Reply:
It doesn't make sense to have status and exception too, for me it feels the same thing with different solutions in the same method. Please refactor for infrastructure too

Reply:
I've committed the changes